Failed to build qtbase for target i686-w64-mingw32.static.posix

Hi,

I have been trying to build mxe to cross-compile sigrok ([link](https://github.com/sigrokproject/sigrok-util/tree/master/cross-compile/mingw)) on Ubuntu. 

The make command used is as per the README in the sigrok-util/cross-compile/mingw/ repo ([link](https://github.com/sigrokproject/sigrok-util/blob/master/cross-compile/mingw/README))

It is as follows: 

``` 
make -j4 MXE_TARGETS=i686-w64-mingw32.static.posix \   
MXE_PLUGIN_DIRS=plugins/examples/qt5-freeze \   
gcc glib libzip libusb1 libftdi1 hidapi glibmm qtbase qtimageformats \   
qtsvg qttranslations boost check gendef libieee1284 \   
qtbase_CONFIGURE_OPTS='-no-sql-mysql' --keep-going
```
I added the ```-j4``` and ```--keep-going```, but the error I am seeing is the same without these changes too. 

I am getting the following error - 
```
[build]       postgresql               i686-w64-mingw32.static.posix
[done]        postgresql               i686-w64-mingw32.static.posix                    231840 KiB     1m29.627s
[build]       sqlite                   i686-w64-mingw32.static.posix
[done]        sqlite                   i686-w64-mingw32.static.posix                    16400 KiB      0m8.741s
[build]       qtbase                   i686-w64-mingw32.static.posix

Failed to build package qtbase for target i686-w64-mingw32.static.posix!
------------------------------------------------------------
make[2]: *** [sub-src-make_first] Error 2
make[2]: Target 'first' not remade because of errors.
make[2]: Leaving directory '/home/user/mxe-git/tmp-qtbase-i686-w64-mingw32.static.posix/qtbase-opensource-src-5.7.1'
Makefile:855: recipe for target 'build-only-qtbase_i686-w64-mingw32.static.posix' failed
make[1]: *** [build-only-qtbase_i686-w64-mingw32.static.posix] Error 2
make[1]: Leaving directory '/home/user/mxe-git'
real	2m35.354s
user	6m57.056s
sys	0m34.564s
------------------------------------------------------------
[log]      /home/user/mxe-git/log/qtbase_i686-w64-mingw32.static.posix

Makefile:855: recipe for target '/home/user/mxe-git/usr/i686-w64-mingw32.static.posix/installed/qtbase' failed
make: *** [/home/user/mxe-git/usr/i686-w64-mingw32.static.posix/installed/qtbase] Error 1
make: Target 'qtbase' not remade because of errors.
```

i was not able to understand the root cause by looking at the log file (available at https://gist.github.com/annr9/745b377d8d14e2c88f69a01ca0653328) . 

Please point me to any mistake I may be making in the process or something I may be missing in the log file. Also let me know if you need any further details from me. 

Thanks,
annr

I think thats just the configure check. There are multiple of it when you follow along the log. Failed compile means feature gets disabled.

Actual error is far later, starting line 1363
```
...
io/qfilesystemengine_win.cpp:634:16: error: redefinition of 'struct _FILE_ID_INFO'
...
io/qfilesystemengine_win.cpp:637:3: error: conflicting declaration 'typedef int FILE_ID_INFO'
...
io/qfilesystemengine_win.cpp:637:18: error: conflicting declaration 'typedef int* PFILE_ID_INFO'
...
Makefile.Release:23485: recipe for target '.obj/release/qfilesystemengine_win.o' failed
make[5]: *** [.obj/release/qfilesystemengine_win.o] Error 1
```

I can't tell you though why it is failing there.

--- a/src/gcc.mk
+++ b/src/gcc.mk
@@ -70,11 +70,11 @@ define $(PKG)_BUILD_mingw-w64
     cd '$(BUILD_DIR).headers' && '$(BUILD_DIR)/$(mingw-w64_SUBDIR)/mingw-w64-headers/configure' \
         --host='$(TARGET)' \
         --prefix='$(PREFIX)/$(TARGET)' \
-        --with-default-msvcrt=msvcrt \
         --enable-sdk=all \
         --enable-idl \
         --enable-secure-api \
         --with-default-msvcrt=msvcrt \
+        --with-default-win32-winnt=0x0600 \
         $(mingw-w64-headers_CONFIGURE_OPTS)
     $(MAKE) -C '$(BUILD_DIR).headers' install
 
@@ -89,6 +89,7 @@ define $(PKG)_BUILD_mingw-w64
         --host='$(TARGET)' \
         --prefix='$(PREFIX)/$(TARGET)' \
         --with-default-msvcrt=msvcrt \
+        --with-default-win32-winnt=0x0600 \
         @gcc-crt-config-opts@ \
         $(mingw-w64-crt_CONFIGURE_OPTS)
     $(MAKE) -C '$(BUILD_DIR).crt' -j '$(JOBS)' || $(MAKE) -C '$(BUILD_DIR).crt' -j '$(JOBS)'

